mantidproject/mantid
# Engineering Diffraction GUI bank combo box removed and assorted refactoring

After the work in #21164, the combo box for bank number in the ED GUI was made obsolete, as the list widget takes care of both run and bank numbers. This PR took care of removing the combo box and all related code.

A large amount of refactoring was also done, which involved moving some helper functions to anonymous namespaces (instead of having them as methods) and renaming some methods and attributes whose purpose did not match their original name, 

Addresses #21238 

**To test:**

Essentially, make sure everything still works as it did after #21226.

Some sample data is [here](https://github.com/mantidproject/mantid/files/1460806/ENGINX_focused_runs.zip).

Load a run from the above data by browsing with the first **Browse** button, then clicking **Load**.
Select the loaded run from the list widget by clicking its label - it should be plotted in the plot window.
Click **Fit** to do a fit - some workspaces with fit parameters should appear in the Workspaces area, and the fitted peaks should be plotted over the original plot. **Update:** The x axis label should be updated when workspaces are plotted so that the workspace units and plot label actually match.

Try changing the peaks with the Peaks line-edit, fitting while multiple runs are loaded, and anything else you think might break things.

Check that fitting information was saved (as `.csv`). On Windows this will be in `C:\EnginX_Mantid\User`, on Linux `~/EnginX_Mantid/User`.

**Release Notes**
The changes to the fitting tab are not complete - this change is a small part of the effort to enable multi-run fitting, which will be included in the release notes once it's been put to bed.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
